File: composeApp/src/androidMain/kotlin/com/zrifapps/goservice/ui/vehicle/components/ComponentVisuals.kt
package com.zrifapps.goservice.ui.vehicle.components

import androidx.compose.ui.graphics.Color
import com.zrifapps.goservice.feature.component.domain.model.Component
import com.zrifapps.goservice.feature.component.domain.model.ComponentInterval
import com.zrifapps.goservice.feature.component.domain.model.ComponentTag as DomainComponentTag
import com.zrifapps.goservice.feature.vehicle.domain.model.VehicleType
import com.zrifapps.goservice.ui.theme.FaIcons

fun Component.faIcon(): String = when (iconKey) {
    "OIL_CAN" -> FaIcons.OIL_CAN
    "BOLT" -> FaIcons.BOLT
    "CAR_BATTERY" -> FaIcons.CAR_BATTERY
    "CIRCLE_NOTCH" -> FaIcons.CIRCLE_NOTCH
    "LIFE_RING" -> FaIcons.LIFE_RING
    "FILTER" -> FaIcons.FILTER
    "TEMPERATURE_HALF" -> FaIcons.TEMPERATURE_HALF
    "GEAR" -> FaIcons.GEAR
    "GEARS" -> FaIcons.GEARS
    "WRENCH" -> FaIcons.WRENCH
    else -> FaIcons.WRENCH
}

fun Component.uiColor(): Color = parseHexColorOrDefault(colorHex)

fun Component.intervalLabelFor(vehicleType: VehicleType): String =
    intervalFor(vehicleType)?.displayLabel ?: "—"

fun DomainComponentTag.label(): String = when (this) {
    DomainComponentTag.Core -> "Wajib"
    DomainComponentTag.Plus -> "Umum"
    DomainComponentTag.Pro -> "Opsional"
}

fun ComponentInterval.summary(): String = displayLabel

private fun parseHexColorOrDefault(hex: String, fallback: Color = Color(0xFF5C6357)): Color =
    try {
        Color(android.graphics.Color.parseColor(hex))
    } catch (_: Throwable) {
        fallback
    }

File: iosApp/iosApp/Features/Vehicle/Components/ComponentVisuals.swift
import SwiftUI
import Shared

extension Shared.Component {
    var iconUnicode: String {
        switch iconKey {
        case "OIL_CAN": return "\u{f613}"
        case "BOLT": return "\u{f0e7}"
        case "CAR_BATTERY": return "\u{f5df}"
        case "CIRCLE_NOTCH": return "\u{f1ce}"
        case "LIFE_RING": return "\u{f1cd}"
        case "FILTER": return "\u{f0b0}"
        case "TEMPERATURE_HALF": return "\u{f2c9}"
        case "GEAR": return "\u{f013}"
        case "GEARS": return "\u{e5fe}"
        case "WRENCH": return "\u{f0ad}"
        default: return "\u{f0ad}"
        }
    }

    var uiColor: Color {
        parseHexColor(colorHex) ?? Color(red: 0.36, green: 0.39, blue: 0.34)
    }

    func intervalLabel(for vehicleType: VehicleType) -> String {
        intervalFor(type: vehicleType)?.displayLabel ?? "—"
    }
}

extension Shared.ComponentTag {
    var displayLabel: String {
        switch self {
        case .core: return "Wajib"
        case .plus: return "Umum"
        case .pro: return "Opsional"
        default: return "Umum"
        }
    }
}

private func parseHexColor(_ hex: String) -> Color? {
    let trimmed = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
    var int: UInt64 = 0
    guard Scanner(string: trimmed).scanHexInt64(&int) else { return nil }
    let r = Double((int >> 16) & 0xFF) / 255
    let g = Double((int >> 8) & 0xFF) / 255
    let b = Double(int & 0xFF) / 255
    return Color(red: r, green: g, blue: b)
}
